app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -58,17 +58,17 @@ object BitcoindRpcBackendUtil extends BitcoinSLogger {
       _ <- walletStateOpt match {
         case None =>
           for {
-            utxos <- wallet.listUtxos()
-            lastConfirmedOpt = utxos.filter(_.blockHash.isDefined).lastOption
+            txDbs <- wallet.listTransactions()
+            lastConfirmedOpt = txDbs.filter(_.blockHashOpt.isDefined).lastOption
```

Comment:
Aren't we assuming the database returns things in the correct order here? I don't see how `.lastOpt` guarantees that it is the "tallest" block

Reply:
We are, I am not sure if this is the best, but probably a topic for a separate PR, also this is only ever used if you have an old wallet.